File: app/src/main/java/app/mumandroidassignment/lesson6/byjava/question1/MainActivity.java
package app.mumandroidassignment.lesson6.byjava.question1;

import android.app.Activity;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;

import app.mumandroidassignment.R;

public class MainActivity extends AppCompatActivity {

    private final String FRAGMENT_HOME = "fragment_home";
    private final String FRAGMENT_ABOUT = "fragment_about";
    private final String FRAGMENT_WORK = "fragment_work";
    private final String FRAGMENT_CONTACT = "fragment_contact";

    private final Fragment fragmentHome = new HomeFragment();
    private final Fragment fragmentAbout = new AboutFragment();
    private final Fragment fragmentWork = new WorkFragment();
    private final Fragment fragmentContact = new ContactFragment();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        ActionBar actionBar = getSupportActionBar();
        actionBar.setTitle("Curriculum Vitae");
        getSupportFragmentManager().beginTransaction().add(R.id.root, fragmentHome, FRAGMENT_HOME).commit();
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_lesson6_1, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.mi_home:
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.root, fragmentHome, FRAGMENT_HOME)
                        .commit();
                break;
            case R.id.mi_Work:
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.root, fragmentWork, FRAGMENT_WORK)
                        .commit();
                break;
            case R.id.mi_contact:
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.root, fragmentContact, FRAGMENT_CONTACT)
                        .commit();
                break;
            case R.id.mi_about_me:
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.root, fragmentAbout, FRAGMENT_ABOUT)
                        .commit();
                break;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/app/mumandroidassignment/lesson6/byKotlin/question1/MainActivity.kt
package app.mumandroidassignment.lesson6.byKotlin.question1

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem

import app.mumandroidassignment.R

class MainActivity : AppCompatActivity() {

    private val FRAGMENT_HOME = "fragment_home"
    private val FRAGMENT_ABOUT = "fragment_about"
    private val FRAGMENT_WORK = "fragment_work"
    private val FRAGMENT_CONTACT = "fragment_contact"

    private val fragmentHome by lazy { HomeFragment() }
    private val fragmentAbout by lazy { AboutFragment() }
    private val fragmentWork by lazy { WorkFragment() }
    private val fragmentContact by lazy { ContactFragment() }



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        supportFragmentManager.beginTransaction().add(R.id.root, fragmentHome , FRAGMENT_HOME).commit()
    }


    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_lesson6_1, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        Log.d("zhq.debug","onOptionsItemSelected")
        when (item!!.itemId) {
            R.id.mi_home -> {
                Log.d("zhq.debug","mi_home")
                supportFragmentManager.beginTransaction()
                        .replace(R.id.root, fragmentHome , FRAGMENT_HOME)
                        .commit()
            }
            R.id.mi_about_me ->{
                supportFragmentManager.beginTransaction()
                        .replace(R.id.root, fragmentAbout , FRAGMENT_ABOUT)
                        .commit()
            }
            R.id.mi_Work ->{
                supportFragmentManager.beginTransaction()
                        .replace(R.id.root, fragmentWork , FRAGMENT_WORK)
                        .commit()
            }
            R.id.mi_contact ->{
                supportFragmentManager.beginTransaction()
                        .replace(R.id.root, fragmentContact , FRAGMENT_CONTACT)
                        .commit()
            }
        }
        return super.onOptionsItemSelected(item)
    }
}
